modules/jfx.incubator.richtext/src/main/java/jfx/incubator/scene/control/richtext/CodeArea.java

```diff
@@ -0,0 +1,425 @@
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package jfx.incubator.scene.control.richtext;
+
+import java.util.List;
+import javafx.beans.property.BooleanProperty;
+import javafx.beans.property.DoubleProperty;
+import javafx.beans.property.IntegerProperty;
+import javafx.beans.property.ObjectProperty;
+import javafx.beans.property.SimpleBooleanProperty;
+import javafx.beans.value.WritableValue;
+import javafx.css.CssMetaData;
+import javafx.css.FontCssMetaData;
+import javafx.css.StyleOrigin;
+import javafx.css.Styleable;
+import javafx.css.StyleableDoubleProperty;
+import javafx.css.StyleableIntegerProperty;
+import javafx.css.StyleableObjectProperty;
+import javafx.css.StyleableProperty;
+import javafx.css.converter.SizeConverter;
+import javafx.scene.AccessibleAttribute;
+import javafx.scene.Node;
+import javafx.scene.control.Labeled;
+import javafx.scene.text.Font;
+import com.sun.jfx.incubator.scene.control.richtext.Params;
+import com.sun.jfx.incubator.scene.control.richtext.util.RichUtils;
+import jfx.incubator.scene.control.richtext.model.CodeTextModel;
+import jfx.incubator.scene.control.richtext.model.StyledTextModel;
+import jfx.incubator.scene.control.richtext.skin.CodeAreaSkin;
+import jfx.incubator.scene.control.richtext.skin.RichTextAreaSkin;
+
+/**
+ * CodeArea is an editable text component which supports styling (syntax highlighting) of plain text.
+ * <p>
+ * Unlike its base class {@link RichTextArea}, the {@code CodeArea} requires a special kind of model to be used,
+ * a {@link CodeTextModel}.
+ * <h2>Creating a CodeArea</h2>
+ * <p>
+ * The following example creates an editable control with the default {@link CodeArea}:
+ * <pre>{@code    CodeArea codeArea = new CodeArea();
+ *   codeArea.setWrapText(true);
+ *   codeArea.setLineNumbersEnabled(true);
+ *   codeArea.setText("Lorem\nIpsum");
+ * }</pre>
+ * Which results in the following visual representation:
+ * <p>
+ * <img src="doc-files/CodeArea.png" alt="Image of the CodeArea control">
+ * </p>
+ */
+public class CodeArea extends RichTextArea {
```

Comment:
The class docs could use to be expanded a little, especially regarding the superclass methods that behave differently (e.g., the style is ignored when calling any mutator method that accepts a a style) or superclass properties that are bound / set (e.g. `leftDecorator`).